apps/angular/projection/src/app/ui/card/card.component.ts

```diff
@@ -1,34 +1,45 @@
-import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
-import { ListItemComponent } from '../list-item/list-item.component';
+import { NgFor, NgTemplateOutlet } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ContentChild,
+  EventEmitter,
+  Input,
+  Output,
+  TemplateRef,
+} from '@angular/core';
 
 @Component({
   selector: 'app-card',
-  templateUrl: './card.component.html',
   standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
+  imports: [NgFor, NgTemplateOutlet],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  host: {
+    class: 'border-2 border-black rounded-md p-4 w-fit flex flex-col gap-3',
+  },
+  template: `
+    <ng-content select="img" />
+    <section>
+      <ng-container *ngFor="let item of list">
+        <ng-container
+          [ngTemplateOutlet]="templateRef"
+          [ngTemplateOutletContext]="{ $implicit: item }" />
+      </ng-container>
+    </section>
+    <button
+      class="border border-blue-500 bg-blue-300 p-2 rounded-sm"
+      (click)="addNewItem()">
+      Add
+    </button>
+  `,
 })
-export class CardComponent {
-  @Input() list: any[] | null = null;
-  @Input() type!: CardType;
-  @Input() customClass = '';
+export class CardComponent<T> {
+  @Input() list: T[] | null = null;
+  @Output() add: EventEmitter<void> = new EventEmitter<void>();
+  @ContentChild('tmplRow', { read: TemplateRef })
+  templateRef!: TemplateRef<{ $implicit: T }>;
```

Comment:
you can improve it a bit by using a directive instead of a magic string

Reply:
yes, very agree 🙂